### PR TITLE
feat: New Icon component 

### PR DESCRIPTION
**Pull request by charmi-v**
## Description

Component: Icon

- Introduced a new Icon component.
- Added a corresponding test file.
- Updated existing mock files to support Jest testing.
- Included a documentation file detailing the component and listing the most commonly used icons.

Icon-Docs

## Why

Introduced the new "Icon" component to enhance the shared-component library. 
- This addition allows you to customize icon sizes and select from predefined states. 
- Use this component to maintain consistency and flexibility in icon usage across projects.

## Issue

#235 

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by charmi-v** [comment]
@jjeroch @oyo  Could you please review the dependency issue here and advise on possible resolutions?

I have added "@testing-library/jest-dom": "^6.4.8" to the `devDependencies`.

**Comment by charmi-v** [comment]
> @charmi-v in your given screenshots all icons have the same size container. if a 8px icon is placed, would it still be possible to adjust the size of the container? Imagine there is a small 14px height button which requires a small icon – would the (assuming) 40px container of the icon create an issue?

The container in the screenshots is for illustration only and will not be rendered with the Icon component. Only the icon itself will be rendered, so you can adjust its size without any issues related to the container.

File: package.json [hunk 1]
```diff
@@ -63,10 +63,11 @@
     "@storybook/addon-links": "^8.1.6",
     "@storybook/addon-onboarding": "^8.1.6",
     "@storybook/blocks": "^8.1.6",
+    "@storybook/builder-vite": "^8.1.10",
     "@storybook/react": "^8.1.6",
     "@storybook/react-vite": "^8.1.6",
     "@storybook/test": "^8.1.6",
-    "@storybook/builder-vite": "^8.1.10",
+    "@testing-library/jest-dom": "6.4.5",
```
Comment (oyo) [inline package.json:70]: Why do we add fixed version `6.4.5` here - while `6.4.8` got published some weeks ago. Are there any issues with that?

**Review by oyo — COMMENTED** [review]
@charmi-v can you explain why we are using an old library version?

File: package.json (hunk 1 above)
Comment (charmi-v) [reply package.json:70]: When I created the PR, the Automated GitHub Workflow Action `Check Dependencies / check-dependencies` failed due to the latest version of `@testing-library/jest-dom`. 

Please advise on how to proceed with directly modifying the dependencies.

Comment (oyo) [reply package.json]: okay, updated your branch and it seems resolved now

**Review by oyo — APPROVED** [review]

**Comment by ybidois** [comment]
Great work @charmi-v! 👏

**Comment by MaximilianHauer** [comment]
woah ! thank you for your support 👍

**Pull request merged**